lib/matplotlib/axes/_base.py

```diff
@@ -1060,6 +1060,10 @@ def cla(self):
             self.yaxis.set_visible(yaxis_visible)
             self.patch.set_visible(patch_visible)
         self.stale = True
+        if(self.xaxis_inverted()):
```

Comment:
I think the way to do this is in each of the "else" blocks of the "if sharex", sharey conditionals, add `self.viewLim.intervalx = (0, 1)` etc.  The actual values won't matter, only the fact that they are in the normal order.